lptree.pas

```diff
@@ -4262,6 +4272,24 @@ function TLapeTree_Operator.resType(Restructure: Boolean): TLapeType;
 
   Result := inherited resType();
 
+  if (FResType = nil) and (FLeft <> nil) and (FRight <> nil) and (LeftType.BaseType <> ltUnknown) and
```

Comment:
Sure.. but as long as FLeft and FRight is not `nil` there is no reason for LeftType/RightType to be nil. If that happened it would crash elsewhere by the looks of it.